### PR TITLE
Improve mouse cursor handling.

### PR DESCRIPTION
**Pull request by PJB3005**
Added all the standard shapes supported by GLFW.
Fixed memory leak where GLFW cursors weren't being destroyed.
Fixed bug where MouseCursor on window defaulted to null, not Default.

### CODE REVIEW

File: src/Windowing/OpenToolkit.Windowing.Desktop/NativeWindow.cs [hunk 1]
```diff
@@ -42,6 +42,10 @@ public class NativeWindow : INativeWindow
 
         private KeyboardState _keyboardState = default;
 
+        // Current cursor we assigned to the GLFW window.
+        // Null if the cursor is default.
```
Comment (varon) [inline src/Windowing/OpenToolkit.Windowing.Desktop/NativeWindow.cs]: Later on we document this as "cursor cannot be null". Even though there's a good reason why, it does read as a little inconsistent.

Any ideas on how to resolve this?

File: src/Windowing/OpenToolkit.Windowing.Desktop/NativeWindow.cs [hunk 2]
```diff
@@ -1525,5 +1546,26 @@ private static Common.InputAction MapGlfwInputAction(InputAction action)
...
+                case MouseCursor.StandardShape.Hand:
+                    return CursorShape.Hand;
+                case MouseCursor.StandardShape.HResize:
+                    return CursorShape.HResize;
+                case MouseCursor.StandardShape.VResize:
+                    return CursorShape.VResize;
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(shape), shape, null);
+            }
+        }
```
Comment (varon) [inline src/Windowing/OpenToolkit.Windowing.Desktop/NativeWindow.cs]: Do we need to do this mapping and/or have two separate types?

File: stylecop.ruleset [hunk 3]
```diff
@@ -154,7 +154,7 @@
     <!-- Documentation rules -->
     <Rule Id="SA1600" Action="Error" /> <!-- Elements must be documented -->
     <Rule Id="SA1601" Action="None" Intentional="true" /> <!-- Partial elements must be documented -->
-    <Rule Id="SA1602" Action="Error" /> <!-- Enumeration items must be documented -->
+    <Rule Id="SA1602" Action="None" /> <!-- Enumeration items must be documented -->
```
Comment (varon) [inline stylecop.ruleset]: This is a good change.

File: src/Windowing/OpenToolkit.Windowing.Desktop/NativeWindow.cs (hunk 2 above)
Comment (PJB3005) [reply src/Windowing/OpenToolkit.Windowing.Desktop/NativeWindow.cs]: The alternative would be a massive if else chain of `cursor == MouseCursor.IBeam`, so I'm not sure that's preferable.

**Pull request merged**